Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -57,6 +59,17 @@ namespace ROS2
             // Found the component!
             return component;
         }
+
+        //! Check whether the entity has a component of the given type
+        //! @param entity pointer to entity
+        //! @param typeId type of the component
+        //! @returns true if entity has component with given type
+        bool CheckIfEntityHasComponentOfType(const AZ::Entity* entity, const AZ::Uuid typeId)
+        {
+            auto components = AZ::EntityUtils::FindDerivedComponents(entity, typeId);
+            return !components.empty();
```

Comment:
question: Just wanted to sanity check does this check the whole entity hierarchy or just the entity being passed in? Might be good to clarify just in case.

Also this is super minor but for API comments we general phrase them in such a way that it describes what it does not what it's doing (as opposed to an implementation comment). So this API comment above would simply read `Checks whether the entity has a component of the given type.` (adding one 's' 😅 I know, but in my mind it does make a world of difference - can totally do this in another PR though, I'm happy to start a build now).

Reply:
Sanity: I am checking components belonging to one entity only.
API Comment: totally agree!